drivers/hmis/app/models/hmis/form/instance.rb

```diff
@@ -18,25 +17,46 @@ class Hmis::Form::Instance < ::GrdaWarehouseBase
 
   scope :for_projects, -> { where(entity_type: Hmis::Hud::Project.sti_name) }
   scope :for_organizations, -> { where(entity_type: Hmis::Hud::Organization.sti_name) }
-  scope :for_project_types, -> { where(entity_type: 'ProjectType') }
-  scope :defaults, -> { where(entity_type: nil, entity_id: nil) }
+  scope :defaults, -> { where(entity_type: nil, entity_id: nil, project_type: nil, funder: nil) }
 
+  # Find instances that are for a specific Project
   scope :for_project, ->(project_id) { for_projects.where(entity_id: project_id) }
+
+  # Find instances that are for a specific Organization
   scope :for_organization, ->(organization_id) { for_organizations.where(entity_id: organization_id) }
-  scope :for_project_type, ->(project_type) { for_project_types.where(entity_id: project_type) }
 
+  # Find instances that match a project based on Project Type and Funder
+  scope :for_project_by_type_or_funder, ->(project) do
+    # Project Type matches, or project type is not specified
+    matches_type = fi_t[:project_type].eq(project.project_type).or(fi_t[:project_type].eq(nil))
+
+    funders = project.funders.pluck(:funder).compact
+    matches_funder = if funders.any?
+      # Funder matches, or funder is not specified
+      fi_t[:funder].in(funders).or(fi_t[:funder].eq(nil))
+    else
+      # Funder is not specified
+      fi_t[:funder].eq(nil)
+    end
+
+    funder_or_project_type_exists = fi_t[:funder].not_eq(nil).or(fi_t[:project_type].not_eq(nil))
```

Comment:
I'm probably missing something here, but it looks like you explicitly bring in nil project_type on line 31 and then throw it out here? (same with funder on line 36)

Reply:
Its ok if one of them is nil, but they can't both be nil. That's what the `funder_or_project_type_exists` clause checks for. This scope is only meant to match against instances that specify the project type or the funder.